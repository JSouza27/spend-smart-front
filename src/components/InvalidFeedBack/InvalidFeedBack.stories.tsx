import { ComponentMeta, ComponentStory } from '@storybook/react';
import { ThemeProvider } from 'styled-components';

import InvalidFeedBack from '.';
import theme from '../../styles/theme';

export default {
  title: 'InvalidFeedBack',
  component: InvalidFeedBack,
  args: {
    message: 'Mensagem de error...'
  }
} as ComponentMeta<typeof InvalidFeedBack>;

export const Basic: ComponentStory<typeof InvalidFeedBack> = (args) => (
  <ThemeProvider theme={theme}>
    <InvalidFeedBack {...args} />
  </ThemeProvider>
);
