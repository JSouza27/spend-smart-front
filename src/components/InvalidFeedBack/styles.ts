import styled, { css } from 'styled-components';

export const Wrapper = styled.span`
  ${({ theme }) => css`
    color: ${theme.colors.red_400};
    font-size: ${theme.font.sizes.xsmall};
    line-height: ${theme.font.sizes.medium};
  `}
`;
