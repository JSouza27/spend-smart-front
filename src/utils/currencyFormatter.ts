export class CurrencyFormatter {
  private static language = 'pt-BR';
  private static currency = 'BRL';

  static formatter(num: number): string {
    return (num ?? 0).toLocaleString(this.language, {
      currency: this.currency
    });
  }

  static unformat(num: string): number {
    const nums = num.replace(/[^0-9]+/g, '');
    const i = nums.length - 2;

    return Number(`${nums.substring(0, i)}.${nums.substring(i)}`);
  }
}
