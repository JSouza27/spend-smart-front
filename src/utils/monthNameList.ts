export const monthNameList = [
  'janeiro',
  'fevereiro',
  'mar√ßo',
  'abril',
  'maio',
  'junho',
  'julho',
  'agosto',
  'setembro',
  'outubro',
  'novembro',
  'dezembro'
];
