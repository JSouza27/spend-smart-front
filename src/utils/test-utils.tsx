import { render } from '@testing-library/react';
import React from 'react';
import { ThemeProvider } from 'styled-components';

import theme from '../styles/theme';
import { FormProvider, useForm } from 'react-hook-form';

const AllTheProviders = ({ children }: any) => {
  const methods = useForm();

  return (
    <ThemeProvider theme={theme}>
      <FormProvider {...methods}>{children}</FormProvider>
    </ThemeProvider>
  );
};

const customRender = (
  ui: React.ReactElement<any, string | React.JSXElementConstructor<any>>,
  options?: any
) => render(ui, { wrapper: AllTheProviders, ...options });

export * from '@testing-library/react';
export { customRender as render };
