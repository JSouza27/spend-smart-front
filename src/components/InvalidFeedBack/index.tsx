import React from 'react';

import * as S from './styles';
import { FieldError, FieldErrorsImpl, Merge } from 'react-hook-form';

export type InvalidFeedBackProps = {
  message: string | FieldError | Merge<FieldError, FieldErrorsImpl<any>>;
};

export default function InvalidFeedBack({ message }: InvalidFeedBackProps) {
  return <S.Wrapper>{message.toString()}</S.Wrapper>;
}
