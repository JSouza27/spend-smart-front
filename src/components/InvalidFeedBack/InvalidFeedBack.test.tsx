import { render, screen } from 'utils/test-utils';

import InvalidFeedBack from '.';

describe('<InvalidFeedBack />', () => {
  const errorMessage = 'Essa é um teste de error';
  it('should render the message', () => {
    const { container } = render(<InvalidFeedBack message={errorMessage} />);

    const message = screen.getByText(/essa é um teste de error/i);

    expect(message).toBeInTheDocument();
    expect(message).toHaveStyle({ color: '#D14343' });
    expect(container.firstChild).toMatchSnapshot();
  });
});
