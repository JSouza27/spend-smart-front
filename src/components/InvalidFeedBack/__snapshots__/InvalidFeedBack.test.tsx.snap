// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<InvalidFeedBack /> should render the message 1`] = `
.c0 {
  color: #D14343;
  font-size: 1.2rem;
  line-height: 1.6rem;
}

<span
  class="c0"
>
  Essa Ã© um teste de error
</span>
`;
